modules/calib3d/src/calibration.cpp

```diff
@@ -1529,7 +1628,35 @@ CV_IMPL double cvCalibrateCamera2( const CvMat* objectPoints,
         std::copy(param + 4, param + 4 + 14, k);
 
         if( !proceed )
+        {
+            //do errors estimation
+            if(stdDevs)
+            {
+                Ptr<CvMat> JtJ(cvCreateMat(nparams, nparams, CV_64F));
+                CvMat cvMatM(matM);
+                cvEvaluateJtJ2(JtJ, &matA, &_k, &cvMatM, solver.param, npoints, flags, NINTRINSIC, aspectRatio);
+
+                Mat mask = cvarrToMat(solver.mask);
+                int nparams_nz = countNonZero(mask);
+                Mat JtJinv, JtJN;
+                JtJN.create(nparams_nz, nparams_nz, CV_64F);
+                subMatrix(cvarrToMat(JtJ), JtJN, mask, mask);
+                completeSymm(JtJN, false);
+                cv::invert(JtJN, JtJinv, DECOMP_SVD);
+                double sigma2 = norm(allErrors, NORM_L2SQR) / (total - nparams_nz);
+                Mat stdDevsM = cvarrToMat(stdDevs);
+                int j = 0;
+                for (int s = 0; s < nparams; s++)
+                    if(mask.data[s])
+                    {
+                        stdDevsM.at<double>(s) = std::sqrt(JtJinv.at<double>(j,j)*sigma2);
```

Comment:
sigma2 is the variance of the noise. See this [paper](https://stat.ethz.ch/~geer/bsa199_o.pdf) for details.